### ISSUE TITLE
need automated tests on Writer FSM

### ISSUE THREAD

**Issue by zmichaels11** [ros2/rosbag2#195]
`Writer` utilizes a FSM transitioning between `ctor` -> `open` -> [`create_topic` | `remove_topic` | `write`] -> `close` -> `dtor`

There should be automated tests validating the transitions between these states

See https://github.com/ros2/rosbag2/pull/184#pullrequestreview-306261209 for concern.

**Comment by dabonnie** [comment]
We (Amazon) can work on this issue :-)

**Comment by wjwwood** [comment]
> Hey @wjwwood we (Amazon) can work on this issue :-)

Of course! I'm going to assign it to you so it doesn't come up in our triage again. Thanks!

**Comment by MichaelOrlov** [comment]
- Closing as an outdated issue
